v2/concordium/types.proto

```diff
@@ -0,0 +1,459 @@
+    Delegate delegate = 2;
+  }
+}
+
+// A sequence number that determines the ordering of transactions from the
+// account. The minimum sequence number is 1.
+message Nonce {
+  // The nonce
+  uint64 value = 1;
+}
+
+// An amount of microCCD.
+message Amount {
+  uint64 value = 1;
+}
+
+// The amount of microCCD.
+message CredentialIndex {
+  uint32 value = 1;
+}
+
+// The number of signatures required to sign.
+message SignatureThreshold {
+  uint32 value = 1;
+}
+
+// The number of credentials required to sign an account transaction.
+message AccountThreshold {
+  uint32 value = 1;
+}
+
+// An account encryption key. Always 96 bytes.
+message EncryptionKey {
+  bytes value = 1;
+}
+
+// An address of an account. Always 32 bytes.
+message AccountAddress {
+  bytes value = 1;
+}
+
+// A public key used to verify transaction signatures from an account.
+message AccountVerifyKey {
+  oneof key {
+    bytes ed25519_key = 1;
+  }
+}
+
+// Public keys of a single credential.
+message CredentialPublicKeys {
+  map<uint32, AccountVerifyKey> keys = 1;
+  SignatureThreshold threshold = 2;
+}
+
+// A registration ID of a credential, derived from the secred PRF key and a
+// nonce. This is always 48 bytes long.
+message CredentialRegistrationId {
+  bytes value = 1;
+}
+
+// An index of the identity provider that identifies them uniquely in the
+// context of a specific chain.
+message IdentityProviderIdentity {
+  uint32 value = 1;
+}
+
+// Representation of the pair of a year and month.
+message YearMonth {
+  uint32 year = 1;
+  uint32 month = 2;
+}
+
+// Policy on a credential.
+message Policy {
+  // The year and month when the identity object from which the credential is
+  // derived was created.
+  YearMonth created_at = 1;
+  // The last year and month when the credential is still valid. After this
+  // expires an account can no longer be created from the credential.
+  YearMonth valid_to = 2;
+  // Mapping from attribute tags to attribute values. Attribute tags are always
+  // representable in a single `u8`, attribute values are never more than 31
+  // bytes in length.
+  map<uint32, bytes> attributes = 3;
+}
+
+// Values contained in an initial credential.
+message InitialCredentialValues {
+  // Public keys of the credential.
+  CredentialPublicKeys keys = 1;
+  // Its registration ID.
+  CredentialRegistrationId cred_id = 2;
+  // The identity provider who signed the identity object from which this
+  // credential is derived.
+  IdentityProviderIdentity ip_id = 3;
+  // Policy of this credential.
+  Policy policy = 4;
+}
+
+/// Data relating to a single anonymity revoker sent by the account holder to
+/// the chain.
+message ChainArData {
+  // Share of the encryption of IdCredPub.
+  bytes enc_id_cred_pub_share = 1;
+}
+
+// The number of anonymity revokers needed to revoke anonymity of a credential
+// holder.
+message ArThreshold {
+  uint32 value = 1;
+}
+
+// A single commitment in the G1 group of the BLS curve.
```

Comment:
We should specify the size of this, if we know it.

Reply:
We do, and added.